client/src/main/java/io/pravega/client/segment/impl/SegmentOutputStreamImpl.java

```diff
@@ -184,14 +185,23 @@ private ClientConnection waitForConnection() throws ConnectionFailedException, S
                 throw new RuntimeException(e);
             }
         }
+        
+        private void setEventNumber(long minEventNumber) {
```

Comment:
It is the boundary between event numbers which can go over the wire and those that should be considered acked. So for a new reader it is the lowest event number that can be referred to. I have changed the name to ackLevel for clarity.